### ISSUE TITLE
Child collections cannot be queried by LINQ expressions

### ISSUE THREAD

**Issue by KasperDamgaard** [JasperFx/marten#1707]
Hey

I created a minimal example of the problem as seen below. This problems exists at least in the 4.0.0-alpha.4 release
This functionality works as expected in 3.1.3. 

    ```
            public class Program
    {
        static async Task Main(string[] args)
        {
            var parentItem = new Parent
            {
                Children = new List<Child>
                {
                    new()
                    {
                        Dummy = "dummy"
                    }
                }
            };

            IDocumentStore documentStore =
                DocumentStore.For(
                    "INSERT OWN CONN_STRING HERE");
            
            using (var session = documentStore.OpenSession(SessionOptions.ForCurrentTransaction()))
            {
                session.Store(parentItem);
                await session.SaveChangesAsync();
            }
            
            using (var session = documentStore.OpenSession(SessionOptions.ForCurrentTransaction()))
            {
                await session.Query<Parent>().FirstOrDefaultAsync(p => p.Children.Any(c => c.Dummy == "dummy"));
            }
        }
    }

    public class Parent
    {
        public Guid Id { get; set; }
        public List<Child> Children { get; set; }
    }

    public class Child
    {
        public Guid Id { get; set; }
        public string Dummy { get; set; }
    }
```

Running this program gives the exception: 

WITH mt_temp_id_list1CTE as (
select id, unnest(CAST(ARRAY(SELECT jsonb_array_elements(CAST(d.data ->> 'Children' as jsonb))) as jsonb[])) as data from public.mt_doc_parent as d
)
  , mt_temp_id_list2CTE as (
select id, data from mt_temp_id_list1CTE as d where d.data ->> 'Dummy' = :p0
)
 select d.id, d.data from public.mt_doc_parent as d where id in (select id from mt_temp_id_list2CTE) LIMIT :p1$
$
22023: cannot extract elements from an object
 ---> Npgsql.PostgresException (0x80004005): 22023: cannot extract elements from an object
   at Npgsql.NpgsqlConnector.<ReadMessage>g__ReadMessageLong|194_0(NpgsqlConnector connector, Boolean async, DataRowLoadingMode dataRowLoadingMode, Boolean readingNotifications, Boolean isReadingPrependedMessage)
   at Npgsql.NpgsqlDataReader.NextResult(Boolean async, Boolean isConsuming, CancellationToken cancellationToken)
   at Npgsql.NpgsqlCommand.ExecuteReader(CommandBehavior behavior, Boolean async, CancellationToken cancellationToken)
   at Npgsql.NpgsqlCommand.ExecuteReader(CommandBehavior behavior, Boolean async, CancellationToken cancellationToken)
   at Marten.Services.ManagedConnection.<>c__DisplayClass34_0.<<ExecuteReaderAsync>b__0>d.MoveNext()
--- End of stack trace from previous location ---
   at Marten.Services.ManagedConnection.ExecuteReaderAsync(NpgsqlCommand command, CancellationToken token)
  Exception data:
    Severity: ERROR
    SqlState: 22023
    MessageText: cannot extract elements from an object
    File: jsonfuncs.c
    Line: 2011
    Routine: elements_worker_jsonb

**Comment by jeremydmiller** [comment]
@KasperDamgaard I tried your reproduction steps in master and it worked. Moreover, we have a **lot** of tests against this functionality already that are all passing in CI. 

The next step is finding out what your environment is like. Did you do anything to customize the JSON serialization? That's my first guess as to what's going wrong here.

**Comment by KasperDamgaard** [comment]
@jeremydmiller I am equally confused and embarrassed, as I must have done something strange to my database during all this testing. I tried wiping and restarting it, and it now functions - also the advanced case where I first saw the error. So the error was with the database, not serialization, although that was also a good guess.
I am closing this and never speaking of it again :D - Sorry to waste your time man.
